master/internal/api_experiment.go

```diff
@@ -462,42 +462,31 @@ func (a *apiServer) deleteExperiments(exps []*model.Experiment, userModel *model
 
 	for i, e := range exps {
 		i := i
-
-		wg.Add(1)
-		go func(exp *model.Experiment) {
+		exp := e
+		g.Go(func() error {
 			sema <- struct{}{}
 			defer func() { <-sema }()
-			defer wg.Done()
 
 			agentUserGroup, err := user.GetAgentUserGroup(context.TODO(), *exp.OwnerID, workspaceIDs[i])
 			if err != nil {
 				log.WithError(err).Errorf("failed to delete experiment: %d", exp.ID)
-				return
+				return err
 			}
 
-			checkpoints, err := a.m.db.ExperimentCheckpointsToGCRaw(
-				exp.ID,
-				0,
-				0,
-				0,
-			)
+			checkpoints, err := a.m.db.ExperimentCheckpointsToGCRaw(exp.ID, 0, 0, 0)
 			if err != nil {
 				log.WithError(err).Errorf("failed to delete experiment: %d", exp.ID)
-				return
+				return err
 			}
 
-			if len(checkpoints) > 0 {
-				go func() {
-					err := runCheckpointGCTask(
-						a.m.system, a.m.rm, a.m.db, model.NewTaskID(), exp.JobID, exp.StartTime,
-						taskSpec, exp.ID, exp.Config, checkpoints, []string{fullDeleteGlob},
-						true, agentUserGroup, userModel, nil,
-					)
-					if err != nil {
-						log.WithError(err).Errorf("failed to gc checkpoints for experiment")
-						return
-					}
-				}()
+			err = runCheckpointGCTask(
+				a.m.system, a.m.rm, a.m.db, model.NewTaskID(), exp.JobID, exp.StartTime,
+				taskSpec, exp.ID, exp.Config, checkpoints, []string{fullDeleteGlob},
+				true, agentUserGroup, userModel, nil,
+			)
+			if err != nil {
+				log.WithError(err).Errorf("failed to gc checkpoints for experiment")
+				return err
```

Comment:
almost said nit: https://github.com/uber-go/guide/blob/master/style.md#handle-errors-once

but since the errgroup only captures the first error, this is great.